src/troute-network/troute/DataAssimilation.py

```diff
@@ -62,16 +94,57 @@ def __init__(self, network, from_files, value_dict, da_run=[]):
         self._last_obs_df = pd.DataFrame()
         self._usgs_df = pd.DataFrame()
 
+        usgs_df = pd.DataFrame()
+
         # If streamflow nudging is turned on, create lastobs_df and usgs_df:
         if nudging:
+
             if not from_files:
-                usgs_df = value_dict['usgs_df']
-                #usgs_df = usgs_df.join(network.link_gage_df.reset_index().set_index('gages'),how='inner').set_index('link').sort_index()
-                usgs_df = network.link_gage_df.reset_index().set_index('gages').join(usgs_df).set_index('link').sort_index()
 
-                self._usgs_df = _reindex_link_to_lake_id(usgs_df, network.link_lake_crosswalk)
+                if (legacy_bmi_df):
+
+                    # THIS LINE WAS REPLACED BY THE BMI TRANSPORT STUFF
+                    usgs_df = value_dict['usgs_df']
+
+                else:
+
+                    # check if there are data transported
+                    usgs_Array = value_dict['usgs_Array']
+                    if len(usgs_Array) >0:
+
+                        dateNull = value_dict['dateNull']    
+
+                        datesSecondsArray_usgs = value_dict['datesSecondsArray_usgs']
+                        nDates_usgs = value_dict['nDates_usgs']
+                        stationArray_usgs = value_dict['stationArray_usgs']
+                        stationStringLengthArray_usgs = value_dict['stationStringLengthArray_usgs']
+                        nStations_usgs = value_dict['nStations_usgs']
+                
+                        # Unflatten the arrays
+                        df_raw_usgs = a2df._unflatten_array(usgs_Array,\
+                                    nDates_usgs, nStations_usgs)
+
+                        # Decode time/date axis
+                        timeAxisName = 'time'
+                        freqString = '5T'
+                        df_withDates_usgs = a2df._time_retrieve_from_arrays(\
+                            df_raw_usgs, dateNull, datesSecondsArray_usgs, \
+                            timeAxisName, freqString)
+
+                        # Decode station ID axis
+                        stationAxisName = 'stationId'
+                        usgs_df = a2df._stations_retrieve_from_arrays(\
+                                df_withDates_usgs, stationArray_usgs, \
+                                stationStringLengthArray_usgs, stationAxisName)
+                
+                    #usgs_df = usgs_df.join(network.link_gage_df.reset_index().set_index('gages'),how='inner').set_index('link').sort_index()
+                    usgs_df = network.link_gage_df.reset_index().set_index('gages').join(usgs_df).set_index('link').sort_index()
+
+                    self._usgs_df = _reindex_link_to_lake_id(usgs_df, network.link_lake_crosswalk)
```

Comment:
Should these be un-indented 1 level? So they run regardless of `legacy_bmi_df`?

Reply:
Yes, agreed!